overlord/snapstate/handlers.go

```diff
@@ -2233,16 +2231,14 @@ func (m *SnapManager) undoLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 		return backendErr
 	}
 
-	if err := m.maybeUndoRemodelBootChanges(t); err != nil {
-		return err
-	}
-
```

Comment:
are we sure we can move this one?

Reply:
Not 100%... too risky indeed, I've put it in the previous place and made sure it does not affect the task status.